src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -38,6 +45,89 @@ public bool Invoke (MethodProxy calledMethod, MultiValue instanceValue, IReadOnl
 				methodReturnValue = instanceValue;
 				break;
 
+			//
+			// UnderlyingSystemType
+			//
+			case IntrinsicId.Type_get_UnderlyingSystemType:
+				// This is identity for the purposes of the analysis.
+				methodReturnValue = instanceValue;
+				break;
+
+			case IntrinsicId.Type_GetTypeFromHandle:
+				// Infrastructure piece to support "typeof(Foo)" in IL and direct calls everywhere
+				foreach (var value in argumentValues[0]) {
+					if (value is RuntimeTypeHandleValue typeHandle)
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, new SystemTypeValue (typeHandle.RepresentedType));
+					else if (value is RuntimeTypeHandleForGenericParameterValue typeHandleForGenericParameter)
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, GetGenericParameterValue (typeHandleForGenericParameter.GenericParameter));
+					else if (value == NullValue.Instance)
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, value);
+					else
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, GetMethodReturnValue (calledMethod, returnValueDynamicallyAccessedMemberTypes));
+				}
+				break;
+
+			case IntrinsicId.Type_get_TypeHandle:
+				foreach (var value in instanceValue) {
+					if (value is SystemTypeValue typeValue)
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, new RuntimeTypeHandleValue (typeValue.RepresentedType));
+					else if (value is GenericParameterValue genericParameterValue)
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, new RuntimeTypeHandleForGenericParameterValue (genericParameterValue.GenericParameter));
+					else if (value == NullValue.Instance)
+						methodReturnValue = MultiValueLattice.Meet (methodReturnValue, value);
```

Comment:
The docs say GetTypeHandle throws if the argument is null - should this maybe not add a return value?